### ISSUE TITLE
Bug: [no-unnecessary-condition] doesn't recognize truthy bigints prior to TS 5.0

### ISSUE THREAD

**Issue by kirkwaiblinger** [typescript-eslint/typescript-eslint#10242]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.9.5&fileType=.tsx&code=CYUwxgNghgTiAEYD2A7AzgF3hmBXDAFgJ4BCAlgOYCSKGAXPAIwoDcAUG2QGbwAUO%2BYuWq0AlPADeHAL5A&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tiacTJTIAhtEK0yHJgBNK%2BSpPRQA7iKaRwYAL4hNQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
declare const truthyBigInt: 1n;

if (truthyBigInt) {

}
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-unnecessary-condition": "warn"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

Reported as unnecessary (always-truthy) condition

### Actual Result

not reported when using TS < 5.0. Reported when using TS >= 5.0

### Additional Info

This is a bug that affects all usages of `type.isLiteral()` with `bigints` before TS 5.0 (we have 3 as far as I can tell, so one could probably make a similar repro in `no-redundant-type-constituents` and `no-unnecesssary-type-assertion`), see https://github.com/microsoft/TypeScript/pull/50929 and https://github.com/JoshuaKGoldberg/ts-api-utils/pull/545

**Comment by kirkwaiblinger** [comment]
Now that I'm looking at this again, I'm deciding to `wontfix` this. 

1. It's external, already fixed upstream, and TS 4.8/4.9 are at the very edge of our support window, so will most likely be obsolete by our next major release
2. extremely niche

If someone objects please reopen!
